app/src/main/java/com/lagradost/cloudstream3/MainAPI.kt

```diff
@@ -1193,6 +1193,9 @@ interface LoadResponse {
     var syncData: MutableMap<String, String>
     var posterHeaders: Map<String, String>?
     var backgroundPosterUrl: String?
+    var contentRating: String?
+        get() = throw NotImplementedError()
+        set(value) { contentRating = value }
```

Comment:
Is this WIP?
The setter is recursive and I am unsure how jackson will like exceptions on getters :pensive:

Reply:
> Is this WIP? The setter is recursive and I am unsure how jackson will like exceptions on getters 😔

@Blatzar 

I can just remove the exception and make it return nothing, without the get and set, contentRating is required and the constructors can't avoid it, with this it makes overriding it not required so I did it that why, unless maybe there is another way?

Reply:
The setter will not work, it calls itself and might even crash the app due to infinite recursion.
If you fix the setter to `set(value) { field = value }` (redundant) it will complain with "Property in an interface cannot have a backing field".

Whatever you are doing is not how its supposed to be done language wise, you'll have to do it any other way :pensive: 

As for the getter: Exceptions in general should really really really be avoided in android as that kills the app immediately. If you ever need to return nothing then nullable is the way to go.

If you really want to crash on incorrect usage of something (not recommended) use debugAssert and it will not cause crashes in production if something is wrong.

Reply:
@Blatzar what about like:
```kt
var contentRating: String?
    get() = null
    set(value) { return }
```

Reply:
Nevermind I was not thinking straight, I do not believe this is even necessary

Reply:
I removed it @Blatzar and it still works fine, doing provider tests no longer show the error and they appear to work now with backward compatability still.